### PR TITLE
re-done admin navigation, depending on cities of user, added dropdown…

### PR DESCRIPTION
**Pull request by barangerbenjamin**
… choices

is line 53 in www/app/views/layouts/admin.html.erb ok ? It works but I feel weird calling a city.find there

### CODE REVIEW

**Comment by ssaunier** [comment]
@barangerbenjamin can you merge `v3` in your branch, solve the conflict, and use Cedric's new layout with dynamic cities? Thanks 🙏

**Comment by barangerbenjamin** [comment]
⚠️ Somehow, had to do everything in JS because of the gem chosen rails ⚠️ 
I think we should have a quick call about this one 😱

File: app/assets/javascripts/admin/admin.js [hunk 1]
```diff
@@ -9,4 +9,23 @@
...
+
+  if (window.location.href.split("/")[5] != null) {
+    const activeCity = window.location.href.split("/")[5];
+    document.querySelector(".chosen-single > span").innerText = activeCity;
+  }
+  // listen for a click on a city and redirect towards it
+  let chosenResults = document.querySelector(".chosen-results")
+  chosenResults.addEventListener("click", function(event) {
+    // normalize the city name to make it match the city slug, eg: São Paulo => Sao Paulo
+    let cityName = event.srcElement.innerText.normalize('NFD').replace(/[\u0300-\u036f]/g, "")
```
Comment (ssaunier) [inline app/assets/javascripts/admin/admin.js]: You should build the `select` such as the displayed label is the city name but the **value** is the **slug**.

```html
<select>
  <option value="sao-paulo">São Paulo</option>
</select>
```

This way you won't have to use this horrible regex 😉 and directly use the selected value.

Comment (ssaunier) [reply app/assets/javascripts/admin/admin.js]: What about:

```erb
<option value="<%= city.slug %>"><%= city.name %></option>
```

Comment (ssaunier) [reply app/assets/javascripts/admin/admin.js]: Actually, Rails has a [nice helper](https://apidock.com/rails/v4.2.7/ActionView/Helpers/FormOptionsHelper/options_from_collection_for_select) for that:

```erb
options_from_collection_for_select(@cities, :slug, :name)
```

**Comment by ssaunier** [comment]
⚠️ Conflict, please merge up-to-date `v3` branch into yours 🙏

File: app/assets/javascripts/admin/admin.js (hunk 1 above)
Comment (barangerbenjamin) [reply app/assets/javascripts/admin/admin.js]: I don't think I understand, the helper helps not bypassing the iterations but in the end it's still the name displayed in the dropdown and I don't see how to retrieve the slug from the JS

**Comment by barangerbenjamin** [comment]
@ssaunier Can you  merge the PR 🙏

**Pull request merged**